### PR TITLE
add confirmation option to actions

### PR DESCRIPTION
**Pull request by iantrich**
extracted from https://github.com/home-assistant/home-assistant-polymer/pull/3994

Docs: https://github.com/home-assistant/home-assistant.io/pull/10834

### CODE REVIEW

File: src/panels/lovelace/common/handle-click.ts
Comment (bramkragten) [inline src/panels/lovelace/common/handle-click.ts]: Use template literals.
```suggestion
          `Are you sure you want to ${actionConfig.action}?`
```

File: src/data/lovelace.ts
Comment (bramkragten) [inline src/data/lovelace.ts]: Why the Type?
```suggestion
export interface ConfirmationRestrictionConfig {
```

File: src/data/lovelace.ts
Comment (bramkragten) [inline src/data/lovelace.ts]:
```suggestion
export interface RestrictionConfig {
```

**Comment by iantrich** [comment]
@bramkragten waiting for docs for double tap to merge first: https://github.com/home-assistant/home-assistant.io/pull/10808

**Comment by iantrich** [comment]
If you're good with this, we can merge and I'll get docs done after the double tap's are merged

**Review by bramkragten — APPROVED** [review]

**Pull request merged**
